<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1241</width>
    <height>642</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Vipster</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/vipster.qrc">
    <normaloff>:/images/icon.png</normaloff>:/images/icon.png</iconset>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks|QMainWindow::ForceTabbedDocks|QMainWindow::VerticalTabs</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1,0,0">
    <item>
     <layout class="QHBoxLayout" name="modeLayout">
      <item>
       <widget class="QPushButton" name="camButton">
        <property name="text">
         <string>Camera</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">modeGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="selButton">
        <property name="text">
         <string>Select</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">modeGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="modButton">
        <property name="text">
         <string>Modify</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">modeGroup</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="GLWidget" name="openGLWidget">
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="displayLayout" stretch="0,0,0,0,0">
      <item>
       <widget class="QLabel" name="camLabel">
        <property name="text">
         <string>Align camera:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="alignLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <widget class="QPushButton" name="pxButton">
          <property name="text">
           <string notr="true">+x</string>
          </property>
          <property name="checkable">
           <bool>false</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">cameraGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pyButton">
          <property name="text">
           <string notr="true">+y</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">cameraGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pzButton">
          <property name="text">
           <string notr="true">+z</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">cameraGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="mxButton">
          <property name="text">
           <string notr="true">-x</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">cameraGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="myButton">
          <property name="text">
           <string notr="true">-y</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">cameraGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="mzButton">
          <property name="text">
           <string notr="true">-z</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">cameraGroup</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="displaySpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="multLabel">
        <property name="text">
         <string>Cell multiply:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="multLayout">
        <item>
         <widget class="QSpinBox" name="xMultBox">
          <property name="minimum">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="yMultBox">
          <property name="minimum">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="zMultBox">
          <property name="minimum">
           <number>1</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="stepLayout" stretch="0,0,1,0,0,0">
      <item>
       <widget class="QLabel" name="selectLabel">
        <property name="text">
         <string>Select step:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="controlLayout">
        <item>
         <widget class="QPushButton" name="firstStepButton">
          <attribute name="buttonGroup">
           <string notr="true">stepGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="preStepButton">
          <property name="text">
           <string/>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">stepGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="playButton">
          <property name="text">
           <string/>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">stepGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="nextStepButton">
          <property name="text">
           <string/>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">stepGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="lastStepButton">
          <property name="text">
           <string/>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">stepGroup</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QSlider" name="stepSlider">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="stepEdit">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="dividerLabel">
        <property name="text">
         <string>/</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="stepLabel">
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1241</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuVipster">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="menuFrom_existing_Molecule">
     <property name="title">
      <string>&amp;From existing Molecule</string>
     </property>
     <addaction name="actionCopy_single_Step"/>
     <addaction name="actionCopy_Trajector"/>
    </widget>
    <addaction name="actionNew_Molecule"/>
    <addaction name="menuFrom_existing_Molecule"/>
    <addaction name="actionLoad_Molecule"/>
    <addaction name="actionSave_Molecule"/>
    <addaction name="separator"/>
    <addaction name="actionNew_Parameter_Set"/>
    <addaction name="actionSave_Parameter_set"/>
    <addaction name="separator"/>
    <addaction name="actionScreenshot"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionNew_Atom"/>
    <addaction name="actionDelete_Atom_s"/>
    <addaction name="actionCopy_Atom_s"/>
    <addaction name="actionPaste_Atom_s"/>
    <addaction name="separator"/>
    <addaction name="actionHide_Atom_s"/>
    <addaction name="actionShow_Atom_s"/>
    <addaction name="separator"/>
    <addaction name="actionUndo"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout_Vipster"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <addaction name="menuVipster"/>
   <addaction name="menuEdit"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QDockWidget" name="molDock">
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>&amp;Molecule</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="molDockContents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="MolWidget" name="molWidget" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="paramDock">
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>&amp;Parameter</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="paramDockContents">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="ParamWidget" name="paramWidget" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionNew_Molecule">
   <property name="text">
    <string>&amp;New Molecule</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionLoad_Molecule">
   <property name="text">
    <string>&amp;Load Molecule</string>
   </property>
  </action>
  <action name="actionSave_Molecule">
   <property name="text">
    <string>&amp;Save Molecule</string>
   </property>
  </action>
  <action name="actionNew_Parameter_Set">
   <property name="text">
    <string>New &amp;Parameter set</string>
   </property>
  </action>
  <action name="actionSave_Parameter_set">
   <property name="text">
    <string>Sa&amp;ve Parameter set</string>
   </property>
  </action>
  <action name="actionScreenshot">
   <property name="text">
    <string>S&amp;creenshot</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionNew_Atom">
   <property name="text">
    <string>&amp;New Atom</string>
   </property>
   <property name="shortcut">
    <string>N</string>
   </property>
  </action>
  <action name="actionDelete_Atom_s">
   <property name="text">
    <string>&amp;Delete Atom(s)</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionCopy_Atom_s">
   <property name="text">
    <string>&amp;Copy Atom(s)</string>
   </property>
  </action>
  <action name="actionPaste_Atom_s">
   <property name="text">
    <string>&amp;Paste Atom(s)</string>
   </property>
  </action>
  <action name="actionHide_Atom_s">
   <property name="text">
    <string>&amp;Hide Atom(s)</string>
   </property>
  </action>
  <action name="actionShow_Atom_s">
   <property name="text">
    <string>&amp;Show Atom(s)</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>&amp;Undo</string>
   </property>
  </action>
  <action name="actionAbout_Vipster">
   <property name="text">
    <string>&amp;About Vipster</string>
   </property>
  </action>
  <action name="actionCopy_single_Step">
   <property name="text">
    <string>&amp;Copy single Step</string>
   </property>
  </action>
  <action name="actionCopy_Trajector">
   <property name="text">
    <string>C&amp;opy Trajectory</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About &amp;Qt</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>qt/glwidget.h</header>
   <slots>
    <slot>setMode(int,bool)</slot>
    <slot>setMult(int)</slot>
    <slot>setCamera(int)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>MolWidget</class>
   <extends>QWidget</extends>
   <header>qt/molwidget.h</header>
   <container>1</container>
   <slots>
    <signal>stepChanged()</signal>
    <signal>molChanged()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>ParamWidget</class>
   <extends>QWidget</extends>
   <header>qt/paramwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>camButton</tabstop>
  <tabstop>selButton</tabstop>
  <tabstop>modButton</tabstop>
  <tabstop>firstStepButton</tabstop>
  <tabstop>preStepButton</tabstop>
  <tabstop>playButton</tabstop>
  <tabstop>nextStepButton</tabstop>
  <tabstop>lastStepButton</tabstop>
  <tabstop>stepSlider</tabstop>
  <tabstop>pxButton</tabstop>
  <tabstop>pyButton</tabstop>
  <tabstop>pzButton</tabstop>
  <tabstop>mxButton</tabstop>
  <tabstop>myButton</tabstop>
  <tabstop>mzButton</tabstop>
  <tabstop>xMultBox</tabstop>
  <tabstop>yMultBox</tabstop>
  <tabstop>zMultBox</tabstop>
 </tabstops>
 <resources>
  <include location="../resources/vipster.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stepEdit</sender>
   <signal>valueChanged(int)</signal>
   <receiver>stepSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1206</x>
     <y>630</y>
    </hint>
    <hint type="destinationlabel">
     <x>1162</x>
     <y>624</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout_Vipster</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>558</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew_Molecule</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>newMol()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>571</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew_Atom</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>editAtoms()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>571</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stepSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setStep(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>715</x>
     <y>619</y>
    </hint>
    <hint type="destinationlabel">
     <x>357</x>
     <y>626</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modeGroup</sender>
   <signal>buttonToggled(int,bool)</signal>
   <receiver>openGLWidget</receiver>
   <slot>setMode(int,bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>805</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDelete_Atom_s</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>editAtoms()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>620</x>
     <y>320</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraGroup</sender>
   <signal>buttonClicked(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>setCamera(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>792</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yMultBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>setMult(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1175</x>
     <y>588</y>
    </hint>
    <hint type="destinationlabel">
     <x>1137</x>
     <y>529</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xMultBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>setMult(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1130</x>
     <y>600</y>
    </hint>
    <hint type="destinationlabel">
     <x>1083</x>
     <y>411</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zMultBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>setMult(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1229</x>
     <y>594</y>
    </hint>
    <hint type="destinationlabel">
     <x>1188</x>
     <y>526</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stepSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>stepEdit</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1130</x>
     <y>621</y>
    </hint>
    <hint type="destinationlabel">
     <x>1191</x>
     <y>621</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stepGroup</sender>
   <signal>buttonClicked(QAbstractButton*)</signal>
   <receiver>MainWindow</receiver>
   <slot>stepBut(QAbstractButton*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>620</x>
     <y>320</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nextStepButton</sender>
   <signal>clicked()</signal>
   <receiver>stepEdit</receiver>
   <slot>stepUp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>293</x>
     <y>627</y>
    </hint>
    <hint type="destinationlabel">
     <x>1189</x>
     <y>622</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>preStepButton</sender>
   <signal>clicked()</signal>
   <receiver>stepEdit</receiver>
   <slot>stepDown()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>626</y>
    </hint>
    <hint type="destinationlabel">
     <x>1188</x>
     <y>619</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setMol(int)</slot>
  <slot>setStep(int)</slot>
  <slot>editAtoms()</slot>
  <slot>about()</slot>
  <slot>newMol()</slot>
  <slot>setStep()</slot>
  <slot>setMol()</slot>
  <slot>stepBut(QAbstractButton*)</slot>
 </slots>
 <buttongroups>
  <buttongroup name="stepGroup"/>
  <buttongroup name="cameraGroup"/>
  <buttongroup name="modeGroup"/>
 </buttongroups>
</ui>
